+ source /root/mugen/libs/locallibs/common_lib.sh
++ python3 --version
++ '[' 0 -eq 0 ']'
++ source /root/mugen/libs/locallibs/common_lib_python.sh
+ main
++ type -t post_test
+ '[' -n function ']'
+ trap post_test EXIT INT HUP TERM
+ rpm -qa
+ grep expect
++ type -t config_params
+ '[' -n '' ']'
++ type -t pre_test
+ '[' -n function ']'
+ pre_test
+ LOG_INFO 'Start to prepare the test environment.'
+ message='Start to prepare the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to prepare the test environment.'
Tue Aug  8 11:14:28 2023 - INFO  - Start to prepare the test environment.
+ DNF_INSTALL 'aide gcc'
+ pkgs='aide gcc'
+ node=1
+ '[' -z '' ']'
+ tmpfile=
++ python3 /root/mugen/libs/locallibs/rpm_manage.py install --pkgs 'aide gcc' --node 1 --tempfile ''
+ tmpfile2=/tmp/tmpbdeyo_ni
+ '[' -z '' ']'
+ tmpfile=/tmp/tmpbdeyo_ni
+ sed -i '$a/home DATAONLY' /etc/aide.conf
+ LOG_INFO 'End to prepare the test environment.'
+ message='End to prepare the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to prepare the test environment.'
Tue Aug  8 11:14:55 2023 - INFO  - End to prepare the test environment.
++ type -t run_test
+ '[' -n function ']'
+ run_test
+ LOG_INFO 'Start to run test.'
+ message='Start to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to run test.'
Tue Aug  8 11:14:56 2023 - INFO  - Start to run test.
+ aide --init
Start timestamp: 2023-08-08 11:14:56 +0800 (AIDE 0.18.5)
AIDE successfully initialized database.
New AIDE database written to /var/lib/aide/aide.db.new.gz

Number of entries:	50283

---------------------------------------------------
The attributes of the (uncompressed) database(s):
---------------------------------------------------

/var/lib/aide/aide.db.new.gz
 MD5       : L0kL6opDsoQadp1ZlckMPA==
 SHA1      : ygB5hyS4gYTR/sD+OOL0fR05gyA=
 SHA256    : qq8cmezMHcWtxKwfs6gohNgVZoDTuBqM
             zU+FwJ4JAyc=
 SHA512    : 9PJVo+UljP392td92/I7AwZKNcDjEYsS
             tastO0ufzzoedLrWHG6JYdsqEaq53E5J
             BLD7HKS++4SgFm0bRPrbJg==
 RMD160    : 6OVclo5aC7L9ft8snfc8HJmARmQ=
 TIGER     : DLEdnOYXzd+vTLSKDs2S0E0mWujAWTol
 CRC32     : ENDzMA==
 WHIRLPOOL : BNot/eC++TPWLcbt0nNr8BmCD5U+4/ZS
             yLsBZPWvaAnO6NdngdGAqRoHekFxz/br
             V6o6hIeuWray1vaQtFmyfQ==
 GOST      : ieh8cNCM1duEQv8w/r3hei455Ybyrlm2
             PzaQ9zfhXMw=
 STRIBOG256: XnlmsMdTI9EP+9as+EMN43dFZFXG026b
             tfHZ9eTt1sM=
 STRIBOG512: vYWKGPYbwS6oBVd19PDgy03BkoUqH2CB
             JwTeVa7BAftO/SICuaTCFLjQh+OIiWib
             j19/LIpkVgeExc86mKHeHw==
 SM3       : D5mPwUpXSDEaI1Po4JOljLQz9FSvzA6B
             YscLwlwFPS4=


End timestamp: 2023-08-08 11:21:07 +0800 (run time: 6m 11s)
+ CHECK_RESULT 0 0 0 'initialization failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='initialization failed'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ mv /var/lib/aide/aide.db.new.gz /var/lib/aide/aide.db.gz
+ useradd testuser
+ aide -c /etc/aide.conf --update
+ grep -A 10 'Changed entries:'
+ grep /etc/group
f > ... mci.H..  : /etc/group
f < ... mc..H..  : /etc/group-
+ CHECK_RESULT 0 0 0 'Update execution failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Update execution failed'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ test -f /var/lib/aide/aide.db.new.gz
+ CHECK_RESULT 0 0 0 'Update execution failed'
+ actual_result=0
+ expect_result=0
+ mode=0
+ error_log='Update execution failed'
+ exit_mode=0
+ '[' -z 0 ']'
+ '[' 0 -eq 0 ']'
+ test 0x '!=' 0x
+ return 0
+ mv /var/lib/aide/aide.db.new.gz /var/lib/aide/aide.db.gz
+ aide --check
+ grep Changed entries:
grep: entries:: No such file or directory
+ CHECK_RESULT 2 0 1 'Check execution failed'
+ actual_result=2
+ expect_result=0
+ mode=1
+ error_log='Check execution failed'
+ exit_mode=0
+ '[' -z 2 ']'
+ '[' 1 -eq 0 ']'
+ test 2x == 0x
+ return 0
+ LOG_INFO 'End to run test.'
+ message='End to run test.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to run test.'
Tue Aug  8 11:36:50 2023 - INFO  - End to run test.
+ CASE_RESULT 0
+ case_re=0
+ test -z ''
+ test 0 -eq 0
+ LOG_INFO 'succeed to execute the case.'
+ message='succeed to execute the case.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'succeed to execute the case.'
Tue Aug  8 11:36:50 2023 - INFO  - succeed to execute the case.
+ exec_result=
+ exit 0
+ post_test
+ LOG_INFO 'Start to restore the test environment.'
+ message='Start to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'Start to restore the test environment.'
Tue Aug  8 11:36:51 2023 - INFO  - Start to restore the test environment.
+ DNF_REMOVE
+ node=1
+ pkg_list=
+ mode=0
+ [[ -z /tmp/tmpbdeyo_ni ]]
+ '[' 0 -ne 0 ']'
+ '[' 1 == 0 ']'
+ python3 /root/mugen/libs/locallibs/rpm_manage.py remove --node 1 --pkgs '' --tempfile /tmp/tmpbdeyo_ni
+ '[' 0 -ne 0 ']'
+ userdel -rf testuser
+ rm -rf testlog /var/log/aide/aide.log /var/lib/aide/aide.db.gz
+ LOG_INFO 'End to restore the test environment.'
+ message='End to restore the test environment.'
+ python3 /root/mugen/libs/locallibs/mugen_log.py --level info --message 'End to restore the test environment.'
Tue Aug  8 11:37:01 2023 - INFO  - End to restore the test environment.
